solidity/contracts/bridge/WalletCoordinator.sol

```diff
@@ -0,0 +1,571 @@
+    }
+
+    /// @notice Updates the value of the depositRefundSafetyMargin parameter.
+    /// @param _depositRefundSafetyMargin New value.
+    /// @dev Requirements:
+    ///      - The caller must be the owner.
+    function updateDepositRefundSafetyMargin(uint32 _depositRefundSafetyMargin)
+        external
+        onlyOwner
+    {
+        depositRefundSafetyMargin = _depositRefundSafetyMargin;
+        emit DepositRefundSafetyMarginUpdated(_depositRefundSafetyMargin);
+    }
+
+    /// @notice Updates the value of the depositSweepMaxSize parameter.
+    /// @param _depositSweepMaxSize New value.
+    /// @dev Requirements:
+    ///      - The caller must be the owner.
+    function updateDepositSweepMaxSize(uint16 _depositSweepMaxSize)
+        external
+        onlyOwner
+    {
+        depositSweepMaxSize = _depositSweepMaxSize;
+        emit DepositSweepMaxSizeUpdated(_depositSweepMaxSize);
+    }
+
+    /// @notice Submits a deposit sweep proposal. Locks the target wallet
+    ///         for a specific time, equal to the proposal validity period.
+    ///         This function does not store the proposal in the state but
+    ///         just emits an event that serves as a guiding light for wallet
+    ///         off-chain members. Wallet members are supposed to validate
+    ///         the proposal on their own, before taking any action.
+    /// @param proposal The deposit sweep proposal
+    /// @param walletMemberContext Optional parameter holding some data allowing
+    ///        to confirm the wallet membership of the caller. This parameter is
+    ///        relevant only when the caller is not a registered proposal
+    ///        submitter but claims to be a member of the target wallet.
+    /// @dev Requirements:
+    ///      - The caller is either a proposal submitter or a member of the
+    ///        target wallet,
+    ///      - The wallet is not time-locked.
+    function submitDepositSweepProposal(
+        DepositSweepProposal calldata proposal,
+        WalletMemberContext calldata walletMemberContext
+    )
+        public
+        onlyProposalSubmitterOrWalletMember(
+            proposal.walletPubKeyHash,
+            walletMemberContext
+        )
+        onlyAfterWalletLock(proposal.walletPubKeyHash)
+    {
+        walletLock[proposal.walletPubKeyHash] =
+            /* solhint-disable-next-line not-rely-on-time */
+            uint32(block.timestamp) +
+            depositSweepProposalValidity;
+
+        emit DepositSweepProposalSubmitted(proposal, msg.sender);
+    }
+
+    /// @notice Wraps `submitDepositSweepProposal` call and reimburses the
+    ///         caller's transaction cost.
+    /// @dev See `submitDepositSweepProposal` function documentation.
+    function submitDepositSweepProposalWithReimbursement(
+        DepositSweepProposal calldata proposal,
+        WalletMemberContext calldata walletMemberContext
+    ) external refundable(msg.sender) {
```

Comment:
Let's confirm in tests this is enough. Sometimes a gas offset needs to be added. [Here](https://github.com/keep-network/tbtc-v2/blob/6ac4d2d237173f85e6124bac90e70adf95bc90cc/solidity/contracts/relay/LightRelayMaintainerProxy.sol#L68-L71) is an example from #579.

Reply:
Here is the gas consumption:
```
·---------------------------------------------------------------------|---------------------------|--------------|-----------------------------·
|                        Solc version: 0.8.17                         ·  Optimizer enabled: true  ·  Runs: 1000  ·  Block limit: 30000000 gas  │
······································································|···························|··············|······························
|  Methods                                                                                                                                     │
······················|···············································|·············|·············|··············|···············|··············
|  Contract           ·  Method                                       ·  Min        ·  Max        ·  Avg         ·  # calls      ·  eur (avg)  │
······················|···············································|·············|·············|··············|···············|··············
|  WalletCoordinator  ·  submitDepositSweepProposal                   ·      37566  ·      65338  ·       48857  ·            7  ·          -  │
······················|···············································|·············|·············|··············|···············|··············
|  WalletCoordinator  ·  submitDepositSweepProposalWithReimbursement  ·          -  ·          -  ·       59796  ·            1  ·          -  │
······················|···············································|·············|·············|··············|···············|··············

```
While the refunded amount is around `32000` as per the assertion mentioned in https://github.com/keep-network/tbtc-v2/pull/575#discussion_r1159631579. 

That is, we need an offset. However, not sure what should be the reference value for the refund?

Reply:
Done in 205add89. Set `25000` as the initial offset value. That gives a refund slightly above `60000` which should exceed the average cost of `submitDepositSweepProposalWithReimbursement` being `59796`